### PR TITLE
Fix typos - 1st Wave

### PR DESCRIPTION
**Pull request by pengwa**
### Description

There are so many typos reported by the review dog, [Optional Lint] actions (example: https://github.com/microsoft/onnxruntime/actions/runs/9864564489/job/27239732367), this PR is to fix some of them.

### CODE REVIEW

File: orttraining/orttraining/core/optimizer/memory_optimizer/recompute_analysis.cc [hunk 1]
```diff
@@ -60,9 +60,9 @@ using OpsetToIgnorableIndicesMap = InlinedHashMap<int, IgnorableInputIndices>;
  * Most recent revisited for ONNX v1.15.0 release - https://github.com/onnx/onnx/blob/b86cc54efce19530fb953e4b21f57e6b3888534c/docs/Operators.md
  *
  * We defined supported list explicitly instead of using a excluding list for the following reasons:
- * 1. Some ops generate indeterministic results (for example using random number generator). We need evaluate whether
+ * 1. Some ops generate in-deterministic results (for example using random number generator). We need evaluate whether
```
Comment (edgchen1) [inline orttraining/orttraining/core/optimizer/memory_optimizer/recompute_analysis.cc:63]:
```suggestion
 * 1. Some ops generate non-deterministic results (for example using random number generator). We need evaluate whether
```

File: orttraining/orttraining/core/optimizer/memory_optimizer/recompute_analysis.cc [hunk 2]
```diff
@@ -60,9 +60,9 @@ using OpsetToIgnorableIndicesMap = InlinedHashMap<int, IgnorableInputIndices>;
...
  *   this is a problem for recompute before adding the support, instead of fixing this after we find and try to
- *   fix convergence issues (which will be very hard if we have multiple indeterministic operators by default supported.)
+ *   fix convergence issues (which will be very hard if we have multiple in-deterministic operators by default supported.)
```
Comment (edgchen1) [inline orttraining/orttraining/core/optimizer/memory_optimizer/recompute_analysis.cc:65]:
```suggestion
 *   fix convergence issues (which will be very hard if we have multiple non-deterministic operators by default supported.)
```

**Review by edgchen1 — COMMENTED** [review]
thanks for cleaning this up!

File: include/onnxruntime/core/framework/op_kernel.h [hunk 3]
```diff
@@ -94,7 +94,7 @@ class OpKernel {
   }
 
   // Override this function to return a list of attributes the session can safely remove
-  // after it is intialized and saved. This option is useful to reduce memory usage
+  // after it is initalized and saved. This option is useful to reduce memory usage
```
[the remark this answers is not in the document]
Comment (pengwa) [reply include/onnxruntime/core/framework/op_kernel.h]: ok looks I need to be careful when using spell check too also.

File: orttraining/orttraining/core/optimizer/compute_optimizer/sceloss_compute_optimization.h [hunk 4]
```diff
@@ -32,10 +32,10 @@ namespace onnxruntime {
  * 2. Its 2nd output (log_prob) MUST NOT be a graph output and MUST NOT be consumed by other nodes.
  * 3. Its ignore_index exists and is a constant scalar value.
  * 4. Its 2nd input label's input node is not a `ShrunkGather` node (to avoid this transformer duplicated applied).
- * 5. Its 2nd input label is 1) a graph input or 2) output of a Reshape node taking a graph input as its data input.
+ * 5. Following PythonOp (FlagAndPrintDensity).
```
Comment (edgchen1) [inline orttraining/orttraining/core/optimizer/compute_optimizer/sceloss_compute_optimization.h:35]: so this is correcting the original comment?

**Review by edgchen1 — APPROVED** [review]

File: orttraining/orttraining/core/optimizer/compute_optimizer/sceloss_compute_optimization.h (hunk 4 above)
Comment (pengwa) [reply orttraining/orttraining/core/optimizer/compute_optimizer/sceloss_compute_optimization.h:35]: yes, this is to update the implementation details.

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**